immutablecollections/_immutableset.py

```diff
@@ -210,9 +213,33 @@ def empty() -> "ImmutableSet[T]":
         """
         return _EMPTY
 
+    def issubset(self, other: Iterable[T]) -> bool:
+        """
+        This set is a subset of another set if all the elements of this set are
+        are contained in the other set.
+
+        Note that this operation is equivalent to `self <= other`. For determining whether or not
+         this set is a _proper_ subset of another set, use `self < other`.
+        """
+        if isinstance(other, Set):
```

Comment:
What does `isinstance(other, Set)` return if `other` is an `ImmutableSet`?

Reply:
Whose bright idea was it to have `collections.abc.Set` be the same as `typing.AbstractSet` and `collections.abc.MutableSet` be `typing.{Mutable,}Set`...

Reply:
| (sub)class                 | ImmutableSet is? | `set` | `frozenset` |
|----------------------------|------------------|-------|-------------|
| typing.AbstractSet         | True             | True  | True        |
| typing.MutableSet          | False            | True  | False       |
| typing.Set                 | False            | True  | False       |
| collections.abc.MutableSet | False            | True  | False       |
| collections.abc.Set        | True             | True  | True        |

Reply:
Looks like `typing.AbstractSet` is the way to go.